source/common/stream_info/filter_state_impl.cc

```diff
@@ -10,16 +10,18 @@ void FilterStateImpl::setData(absl::string_view data_name, std::shared_ptr<Objec
                               FilterState::StreamSharing stream_sharing) {
   if (life_span > life_span_) {
     if (hasDataWithNameInternally(data_name)) {
-      throw EnvoyException(
-          "FilterState::setData<T> called twice with conflicting life_span on the same data_name.");
+      IS_ENVOY_BUG("FilterStateAccessViolation: FilterState::setData<T> called twice with "
```

Comment:
I think this should be OK as long as the `catch` block at the call-sites is not doing repeated attempts. Should we also remove the obvious `try-catch` blocks with `setData` and `getDataMutable`?

Reply:
I looked through and don't see any try/catch around filter state usage in the codebase. Did I miss something?

Reply:
That's great.